index.js

```diff
@@ -1,198 +1,150 @@
-  return isInput(node) && node.type === 'radio';
-}
+  function isRadio(node) {
+    return isInput(node) && node.type === 'radio';
+  }
 
-function isNonTabbableRadio(node) {
-  return isRadio(node) && !isTabbableRadio(node);
-}
+  function isNonTabbableRadio(node) {
+    return isRadio(node) && !isTabbableRadio(node);
+  }
 
-function getCheckedRadio(nodes) {
-  for (var i = 0; i < nodes.length; i++) {
-    if (nodes[i].checked) {
-      return nodes[i];
+  function getCheckedRadio(nodes) {
+    for (var i = 0; i < nodes.length; i++) {
+      if (nodes[i].checked) {
+        return nodes[i];
+      }
     }
   }
-}
-
-function isTabbableRadio(node) {
-  if (!node.name) return true;
-  // This won't account for the edge case where you have radio groups with the same
-  // in separate forms on the same page.
-  var radioSet = node.ownerDocument.querySelectorAll('input[type="radio"][name="' + node.name + '"]');
-  var checked = getCheckedRadio(radioSet);
-  return !checked || checked === node;
-}
-
-// An element is "untouchable" if *it or one of its ancestors* has
-// `visibility: hidden` or `display: none`.
-function UntouchabilityChecker(elementDocument) {
-  this.doc = elementDocument;
-  // Node cache must be refreshed on every check, in case
-  // the content of the element has changed. The cache contains tuples
-  // mapping nodes to their boolean result.
-  this.cache = [];
-}
-
-// getComputedStyle accurately reflects `visibility: hidden` of ancestors
-// but not `display: none`, so we need to recursively check parents.
-UntouchabilityChecker.prototype.hasDisplayNone = function hasDisplayNone(node, nodeComputedStyle) {
-  if (node === this.doc.documentElement) return false;
-
-    // Search for a cached result.
-    var cached = find(this.cache, function(item) {
-      return item === node;
-    });
-    if (cached) return cached[1];
-
-    nodeComputedStyle = nodeComputedStyle || this.doc.defaultView.getComputedStyle(node);
-
-    var result = false;
-
-    if (nodeComputedStyle.display === 'none') {
-      result = true;
-    } else if (node.parentNode) {
-      result = this.hasDisplayNone(node.parentNode);
-    }
-
-    this.cache.push([node, result]);
 
-    return result;
-}
+  function isTabbableRadio(node) {
+    if (!node.name) return true;
+    // This won't account for the edge case where you have radio groups with the same
+    // in separate forms on the same page.
+    var radioSet = node.ownerDocument.querySelectorAll('input[type="radio"][name="' + node.name + '"]');
+    var checked = getCheckedRadio(radioSet);
+    return !checked || checked === node;
+  }
 
-UntouchabilityChecker.prototype.isUntouchable = function isUntouchable(node) {
-  if (node === this.doc.documentElement) return false;
-  var computedStyle = this.doc.defaultView.getComputedStyle(node);
-  if (this.hasDisplayNone(node, computedStyle)) return true;
-  return computedStyle.visibility === 'hidden';
-}
+  function isHidden(node) {
```

Comment:
@davidtheclark This function replaces the UntouchabilityChecker.  All the other changes result from that.
offsetParent tests for display none: https://stackoverflow.com/questions/19669786/check-if-element-is-visible-in-dom



I don't know why the diff is so weird. This is my first pull request on github. Thank you very much for your work.

Reply:
Ohhh, interesting trick! 

I confirmed in Chrome, Firefox, and IE11. Have you confirmed in any other browsers?

It looks to me like you indented the entire file — I wonder if that's causing all the unnecessary diff.

Reply:
Seems like it - https://github.com/davidtheclark/tabbable/pull/32/files?w=1 shows a much shorter diff.